src/town_gui.cpp

```diff
@@ -194,33 +231,15 @@ struct TownAuthorityWindow : Window {
 		switch (widget) {
 			case WID_TA_ACTION_INFO:
 				if (this->sel_index != -1) {
-					SetDParam(0, _price[PR_TOWN_ACTION] * _town_action_costs[this->sel_index] >> 8);
-					DrawStringMultiLine(r.left + WD_FRAMERECT_LEFT, r.right - WD_FRAMERECT_RIGHT, r.top + WD_FRAMERECT_TOP, r.bottom - WD_FRAMERECT_BOTTOM,
-								STR_LOCAL_AUTHORITY_ACTION_TOOLTIP_SMALL_ADVERTISING + this->sel_index);
-				}
-				break;
-			case WID_TA_COMMAND_LIST: {
-				int numact;
-				uint buttons = GetMaskOfTownActions(&numact, _local_company, this->town);
-				int y = r.top + WD_FRAMERECT_TOP;
-				int pos = this->vscroll->GetPosition();
-
-				if (--pos < 0) {
-					DrawString(r.left + WD_FRAMERECT_LEFT, r.right - WD_FRAMERECT_RIGHT, y, STR_LOCAL_AUTHORITY_ACTIONS_TITLE);
-					y += FONT_HEIGHT_NORMAL;
-				}
-
-				for (int i = 0; buttons; i++, buttons >>= 1) {
-					if (pos <= -5) break; ///< Draw only the 5 fitting lines
+					Money action_cost = _price[PR_TOWN_ACTION] * _town_action_costs[this->sel_index] >> 8;
+					bool affordable = action_cost < Company::Get(_local_company)->money;
```

Comment:
This doesn't check that `_local_company` is a valid company.

Reply:
I'd imagine there's a `Company::GetIfValid()` to use here.

Reply:
Using `Company::GetIfValid()` instead of `Company::Get()` in latest commit.
Under what circumstances would `_local_company` not be a valid company? I want to see if I can produce any issues under that situation.